### PR TITLE
feat: add support for conformance testing 'typed' declarative function signature

### PR DESCRIPTION
**Pull request by garethgeorge**
To be used for conformance testing new `typed` declarative function signatures. 

Ruby example command:
```
./client -cmd="bundle exec functions-framework-ruby --source test/conformance/app.rb --target typed_func --signature-type http" --type=http --declarative-type=typed -buildpacks=false
```

In the case of typed functions, a JSON request is generated and it is asserted that the function returns a JSON object that echo's the request in the `payload` field of a new JSON object. I.e. for input `{"a": "b"}` the function should return `{"payload": {"a": "b"}}`.

### CODE REVIEW

**Comment by kenneth-rosario** [comment]
Can you comment an example of how the test functions would look like in any one of the supported languages of your choice for reference?

**Comment by garethgeorge** [comment]
The test function we'll add for ruby is:

```
FunctionsFramework.typed "typed_func" do |request|
  return {
    :payload => request,
  }
end
```

**Review by kenneth-rosario — COMMENTED** [review]
Lets update the README to include new flag as well.

**Comment by garethgeorge** [comment]
Thanks for the initial review Kenneth! Went ahead and updated README.md with examples for the new conformance test type. 

I think I just need another (final this time!) pass to cover the files I've added to update the github actions workflows.

File: README.md [hunk 1]
```diff
@@ -58,6 +58,15 @@ Frameworks to the Functions Framework contract.
...
         ```
+    - **Ruby __typed http__** function Example:
+
+        ```sh
+        $HOME/functions-framework-conformance/client/client \
+          -cmd="bundle exec functions-framework-ruby --source test/conformance/app.rb --target typed_func --signature-type http" \
+          --type=http \
+          --declarative-type=typed \
+          -buildpacks=false
+        ```
```
Comment (kenneth-rosario) [inline README.md:69]: Lets update [the usage table](https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/README.md#usage) and step 1 of the [quick start] (https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/README.md#quickstart)

**Comment by garethgeorge** [comment]
Done, updated flags list.

**Review by kenneth-rosario — APPROVED** [review]
This looks great!

**Pull request merged**